### PR TITLE
Adding linting with Google's rules

### PR DESCRIPTION
**Pull request by TonyBrobston**
I've made modifications to add a style linter using Google's rules. Currently this only runs well at command line, I haven't found an effective manner to run in the IDE. We can deal with this later.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -154,7 +164,34 @@
...
+                            <suppressionsLocation>
+                                checkstyle-suppressions.xml
+                            </suppressionsLocation>
+                            <suppressionsFileExpression>
+                                checkstyle.suppressions.file
+                            </suppressionsFileExpression>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
```
Comment (TonyBrobston) [inline pom.xml]: Above is the main bit to focus on, the rest is a mix of using Intellij's tools to auto format/optimize, with some slight manual tweaking of things. I also ran all tests afterwards.

File: pom.xml [hunk 2]
```diff
@@ -154,7 +164,34 @@
...
+                            <configLocation>google_checks.xml</configLocation>
+                            <encoding>UTF-8</encoding>
+                            <consoleOutput>true</consoleOutput>
+                            <failsOnError>true</failsOnError>
+                            <violationSeverity>warning</violationSeverity>
+                            <failOnViolation>true</failOnViolation>
+                            <suppressionsLocation>
+                                checkstyle-suppressions.xml
+                            </suppressionsLocation>
+                            <suppressionsFileExpression>
```
Comment (AndreasChristianson) [inline pom.xml]: I'm not sure you need this; I think it is related to configuring checkstyle ignored filed via a properties file. If you have `suppressionsLocation`, you might be good.

Comment (TonyBrobston) [reply pom.xml]: I'll take a look at this tomorrow morning.

Comment (TonyBrobston) [reply pom.xml]: Outside of the pom file everything else is fixing linter errors.

Comment (TonyBrobston) [reply pom.xml]: I didn't get to looking at this; probably tomorrow.

**Pull request merged**